### PR TITLE
Add validation for /requests/ api endpoint

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: vumi_http_retry/workers/api/worker.py [hunk 1]
```diff
@@ -42,30 +43,35 @@ def setup(self):
...
     def route_health(self, req):
-        return self.respond(req, {})
+        return response(req, {})
 
     @app.route('/requests/', methods=['POST'])
+    @json_body
+    @validate(
+        has_header('X-Owner-ID'),
+        body_schema({
+            'properties': {
```
Comment (hodgestar) [inline vumi_http_retry/workers/api/worker.py:56]: A few questions about the schema:
- Should we put the schema in a constant and reference it here (that way the schema is separately accessible for testing, etc)?
- Should we  have a `'type': 'object'` at the top-level?
- Should we add a `'items': {'type': 'integer', 'minimum': 0}` to our `intervals`?
- Should we add a `'additionalProperties': {'type': 'array', 'items': 'str'}` to our `headers`?

**Comment by hodgestar** [comment]
Left one comment. Otherwise looks good. Woot on using jsonschema.

File: vumi_http_retry/workers/api/worker.py (hunk 1 above)
Comment (justinvdm) [inline vumi_http_retry/workers/api/worker.py:56]: > Should we put the schema in a constant and reference it here (that way the schema is separately accessible for testing, etc)?

I kind of like it being right next to the route being described, seems more readable. I also don't see us using it as a constant from the tests.

> Should we have a 'type': 'object' at the top-level

+1

> Should we add a 'items': {'type': 'integer', 'minimum': 0} to our intervals

+1

> Should we add a 'additionalProperties': {'type': 'array', 'items': 'str'} to our headers?

+1

Comment (hodgestar) [inline vumi_http_retry/workers/api/worker.py:56]: I'm okay with it staying right next to the route. My thought was that
having it separate helps as validation gets complicated (e.g. one can
re-use it in different places, test validation separately, split out pieces
into separate variables for clarity and re-use). Let's keep it here for now
and adjust later if it feels needed.

Comment (justinvdm) [inline vumi_http_retry/workers/api/worker.py:56]: Sounds good :)

**Comment by justinvdm** [comment]
Ready for re-review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**